Mlem/App/Views/Shared/Images/Helpers/ZoomRecognizer/ZoomRecognizerCoordinator+Logic.swift

```diff
@@ -0,0 +1,307 @@
+            resetMomentum()
+            initialOffset = offset
+            
+            updateOffsetForPanGesture(gesture)
+        case .changed:
+            updateOffsetForPanGesture(gesture)
+        case .ended, .cancelled:
+            panType = .none
+            guard let view = gesture.view else {
+                assertionFailure("Missing view or bounds")
+                return
+            }
+            
+            initialScale = scale
+            
+            let gestureVelocity = gesture.velocity(in: view)
+            let maxOffsets = maxOffsets.compute(scale)
+            let xOob = abs(offset.width) >= maxOffsets.width
+            let yOob = abs(offset.height) >= maxOffsets.height
+            if !(xOob && yOob),
+               abs(gestureVelocity.x) + abs(gestureVelocity.y) > 40 {
+                startMomentum(
+                    velocity: gestureVelocity,
+                    xOob: xOob,
+                    yOob: yOob,
+                    maxXOffset: maxOffsets.width,
+                    maxYOffset: maxOffsets.height
+                )
+            } else {
+                let translation = gesture.translation(in: view)
+                resetToBounds(activeOffset: .init(width: translation.x, height: translation.y).scaled(by: scale))
+            }
+        case .failed:
+            panType = .none
+        default:
+            assertionFailure("Unknown state")
+        }
+    }
+    
+    /// Reacts to the given pan gesture by updating zoom according to the height of the pan gesture
+    func handleZoomPan(gesture: UIPanGestureRecognizer) {
+        switch gesture.state {
+        case .possible:
+            break
+        case .began:
+            initializeBounds(view: gesture.view)
+            guard let bounds else {
+                assertionFailure("No bounds")
+                return
+            }
+            resetMomentum()
+            initialScale = scale
+            initialOffset = offset
+            let xAnchor = (((scale * bounds.width) / 2) - offset.width) / (scale * bounds.width)
+            let yAnchor = (((scale * bounds.height) / 2) - offset.height) / (scale * bounds.height)
+            anchor = .init(x: xAnchor, y: yAnchor)
+        case .changed:
+            let newScale = (initialScale + (gesture.translation(in: nil).y / -60)).bounded(lower: 1.0, upper: 4.0)
+            let maxOffsets = maxOffsets.compute(newScale)
+            let offsetDeltas = computeOffsetDeltas(scaleFactor: newScale / initialScale)
+            let newOffset = initialOffset + offsetDeltas
+            
+            scale = newScale
+            offset = .init(
+                width: newOffset.width.bounded(lower: -maxOffsets.width, upper: maxOffsets.width),
+                height: newOffset.height.bounded(lower: -maxOffsets.height, upper: maxOffsets.height)
+            )
+        case .ended, .cancelled:
+            panType = .none
+        case .failed:
+            panType = .none
```

Comment:
Maybe:

```suggestion
        case .ended, .cancelled, .failed:
            panType = .none
```

(Unless there's a readability reason for it to be as it is)